cclient/serialization/tests/test_json.c

```diff
@@ -713,14 +562,26 @@ void test_deserialize_were_addresses_spent_from(void) {
   init_json_serializer(&serializer);
   const char* json_text =
       "{"
-      "\"states\":[true]"
+      "\"states\":[" BOOL_TRUE "," BOOL_FALSE
+      "]"
       "}";
 
   were_addresses_spent_from_res_t* res = were_addresses_spent_from_res_new();
   serializer.vtable.were_addresses_spent_from_deserialize_response(
       &serializer, json_text, res);
 
-  TEST_ASSERT_TRUE(were_addresses_spent_from_res_at(res, 0));
+  bool* states = NULL;
+  retcode_t ret = were_addresses_spent_from_res_at(res, 0, &states);
+  TEST_ASSERT_EQUAL_INT(RC_OK, ret);
+  TEST_ASSERT_TRUE(*states);
+
+  ret = were_addresses_spent_from_res_at(res, 1, &states);
+  TEST_ASSERT_EQUAL_INT(RC_OK, ret);
+  TEST_ASSERT_FALSE(*states);
+
+  ret = were_addresses_spent_from_res_at(res, 2, &states);
+  TEST_ASSERT_EQUAL_INT(RC_CCLIENT_RES_ERROR, ret);
+  TEST_ASSERT_NULL(states);
```

Comment:
you allocates `states` each time calling `were_addresses_spent_from_res_at` but never freed it.